pop_tools/regrid.py

```diff
@@ -0,0 +1,420 @@
+import os
+import warnings
+
+import cf_xarray
+import numpy as np
+import xarray as xr
+
+try:
+    import xesmf as xe
+except ImportError:
+    message = 'Zonal averaging requires xesmf package.\n\n'
+    'Please conda install as follows:\n\n'
+    ' conda install -c conda-forge xesmf>=0.4.0'
+
+    raise ImportError(message)
+
+from tqdm import tqdm
+
+from .grid import _compute_corners, get_grid
+
+
+def gen_corner_calc(ds, cell_corner_lat='ULAT', cell_corner_lon='ULONG'):
+    """
+    Generates corner information and creates single dataset with output
+    """
+
+    cell_corner_lat = ds[cell_corner_lat]
+    cell_corner_lon = ds[cell_corner_lon]
```

Comment:
I think you want to `assert cell_corner_lon.shape[-1] == 4` to make sure things are in the right order.

Or possibly
```
cell_corner_lat = ds[cell_corner_lat].transpose("nlon", "nlat", ...)
```